src/pkgcheck/checks/python.py

```diff
@@ -687,29 +695,40 @@ def desc(self):
         )
 
 
-class PythonGHDistfileSuffixCheck(Check):
-    """Check ebuilds with PyPI remotes for missing ".gh.tar.gz" suffixes."""
+class PythonInlinePyPIURI(results.VersionResult, results.Warning):
+    """PyPI URI used inline instead of via pypi.eclass"""
+
+    def __init__(self, url: str, replacement: typing.Optional[typing.Tuple[str]] = None, **kwargs):
+        super().__init__(**kwargs)
+        self.url = url
+        self.replacement = replacement
+
+    @property
+    def desc(self) -> str:
+        if self.replacement is None:
+            return ("inline PyPI URI found matching pypi.eclass default, inherit the eclass "
+                    "and remove SRC_URI instead")
+        else:
+            return (f"inline PyPI URI found: {self.url}, inherit pypi.eclass and replace with "
+                    f"$({' '.join(self.replacement)})")
+
+
+class PythonFetchableCheck(Check):
+    """Perform Python-specific checks to fetchables."""
 
     required_addons = (addons.UseAddon,)
-    known_results = frozenset([PythonGHDistfileSuffix])
+    known_results = frozenset([PythonGHDistfileSuffix, PythonInlinePyPIURI])
```

Comment:
Ongoing process, but please replace with set initializer (takes less time, and I think less bytecode, and should be optimized in py3.11 to just use straight frozenset):
```suggestion
    known_results = frozenset({PythonGHDistfileSuffix, PythonInlinePyPIURI})
```